MatrixSDK/Crypto/Dehydration/MXDehydrationService.h

```diff
@@ -0,0 +1,73 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import Foundation;
+
+@class MXCrypto;
+@class MXRestClient;
+@class MXExportedOlmDevice;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// ID of the algorithm used for the dehydration
+FOUNDATION_EXPORT NSString *const MXDehydrationAlgorithm;
+
+/// MXKeyProvider identifier for a 32 bytes long key used to pickle / unpickle the account of the dehydrated device.
+FOUNDATION_EXPORT NSString *const MXDehydrationServiceKeyDataType;
+
+/**
+ Service in charge of dehydrating and rehydrating a device.
+ 
+ @see https://github.com/uhoreg/matrix-doc/blob/dehydration/proposals/2697-device-dehydration.md for more details
+ */
+@interface MXDehydrationService : NSObject
+
+/**
+ Singleton method
+ */
++ (instancetype)sharedInstance;
```

Comment:
Do we really need a shared instance?
The current implementation does not support multi-account well.